gems/sorbet-runtime/lib/types/private/final.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+# typed: false
+
+module T::Private::Final
+  def self.declare(mod)
+    if !mod.is_a?(Module)
+      raise "#{mod.name} is not a module or method and cannot be declared as final"
```

Comment:
Why `or method`?

Reply:
if anything I'd say maybe change the message to say `class or module`.

Reply:
I said "or method" here because you can declare a method as final. I felt like if I said "X is not a module and cannot be declared as final", then that's like saying "if X is not a module, then X cannot be declared as final", which is not true because X could be a method.

I didn't say class because a class is a module. (But maybe people aren't always thinking about that, so I could mention class here too.)